Dockerfile

```diff
@@ -4,7 +4,29 @@
 # Enable BASE_IMAGE as an overrideable ARG for proxy cache + private registry support
 #
 ARG BASE_IMAGE=public.ecr.aws/docker/library/python:3.10-bullseye
-FROM ${BASE_IMAGE}
+FROM ${BASE_IMAGE} as poetry-reqs
+
+ENV PYTHONDONTWRITEBYTECODE 1
+
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install -y \
+        python3-venv && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN python3 -m pip install -U pip poetry
+
+WORKDIR /src
+
+COPY pyproject.toml pyproject.toml
```

Comment:
@climbertjh2 these are single file copies where they're copied in explicitly with the same filename as source (copy `pyproject.toml` _from_ the repo root into the WORKDIR of the container with destination filename of `pyproject.toml`). We want to avoid wildcards to make these layers only repeat when those exact files change.